docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,292 @@
+
+## Motivation
+
+Ensuring the integrity and availability of ETCD backups is crucial for the ability to restore an ETCD cluster after it has irrecoverably gone down. Making backups immutable protects against unintended or malicious modifications post-creation, thereby enhancing the overall security posture.
+
+### Goals
+
+- Implement immutable backup support for ETCD clusters.
+- Secure backup data against unintended or unauthorized modifications after creation.
+- Ensure backups are consistently available and intact for restoration purposes.
+
+### Non-Goals
+
+- Altering existing backup processes beyond what's necessary for immutability.
+- Implementing object-level immutability policies at this stage.
+- Supporting immutable backups on storage providers that do not offer immutability features (e.g., OpenStack Swift).
+
+## Proposal
+
+### Overview
+
+Introduce immutability in backup storage by leveraging cloud provider features that support a write-once-read-many (WORM) model. This will prevent data alterations after backup creation, enhancing data integrity and security. The implementation will focus on bucket-level immutability policies, as they are widely supported and easier to manage across different cloud providers.
+
+### Detailed Design
+
+#### Bucket Immutability Mechanism
+
+The Bucket Immutability feature configures an immutability policy for a cloud storage bucket, governing how long objects in the bucket must be retained in an immutable state. It also allows for locking the bucket's immutability policy, permanently preventing the policy from being reduced or removed.
+
+- **Supported by Major Providers:**
+  - **Google Cloud Storage (GCS):** [Bucket Lock](https://cloud.google.com/storage/docs/bucket-lock)
+  - **Amazon S3 (S3):** [Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html)
+  - **Azure Blob Storage (ABS):** [Immutable Blob Storage](https://learn.microsoft.com/en-us/azure/storage/blobs/immutable-policy-configure-container-scope?tabs=azure-portal)
+
+- **Not Supported:**
+  - **OpenStack Swift**
+
+**Implementation Details:**
+
+- Operators are responsible for creating or updating buckets with these immutability settings before ETCD Druid starts uploading snapshots.
+- Once the bucket is configured to be immutable, snapshots uploaded by `etcd-backup-restore` are also immutable and cannot be altered or deleted until the immutability period expires.
+- No additional configuration needs to be passed to etcd-druid.
+
+#### ETCD Backup Configuration
+
+Operators need to ensure that the ETCD backup configuration aligns with the immutability requirements. This includes setting appropriate immutability periods.
```

Comment:
Can we add a best practice here. Also I believe we have some defaults for garbage collection, so should that be aligned here as well as a guidance?

Reply:
Consumers should not be expected to read the proposal to find defaults. The right place for this is normal documentation.